### ISSUE TITLE
[NEXT] ReferenceInput on Filter: No data when the reference is the same with the resource of the List (i.e. not foreign key)

### ISSUE THREAD

**Issue by afilp** [marmelab/admin-on-rest#1625]
I am using the latest commits on `next`, after BETA-3, as of today.

This problem did not exist up to BETA-3. It was introduced with the recent changes.

**What you were expecting:**
ReferenceInput in Filter to be able to show data, like it used to be in `next` BETA-3.

**What happened instead:**

![image](https://user-images.githubusercontent.com/7850073/37185806-e423d016-234a-11e8-99e1-43be41c845ca.png)

**Steps to reproduce:**
This happens *only* when the reference (`users`) is the of the same resource that the List has. It does not happen for "foreign keys", i.e. when the reference is from other resources. See code below:

**Related code:**
```
// THIS REFERENCEINPUT IS **OK**
//
    <ReferenceInput label="ΡΟΛΟΣ" source="role_id" reference="roles" allowEmpty alwaysOn
                    sort={{field: 'name', order: 'ASC'}}
                    perPage={1000}
    >
      <SelectInput
        optionText="nameGreek"
      />
    </ReferenceInput>

// THIS REFERENCEINPUT IS **NOT OK**, see screenshot above
//
    <ReferenceInput label="ΕΠΩΝΥΜΟ" source="id" reference="users" allowEmpty alwaysOn
                    sort={{field: 'last_name', order: 'ASC'}}
                    perPage={1000}
    >
      <SelectInput optionText="last_name" />
    </ReferenceInput>

```

Based on the UI screenshot above, we should have 3 "references" in `possibleValues`, but we are missing one (the `users@id` one):
![image](https://user-images.githubusercontent.com/7850073/37186863-489bc896-2350-11e8-8d9b-0610ec377169.png)

**Environment**

* Admin-on-rest version:   commit #6d2f265, i.e. latest commit as of today
* Last version that did not exhibit the issue (if applicable):  BETA-3

**Comment by fzaninotto** [comment]
I don't have the model to reproduce the bug, could you try to narrow the commit which caused it using `git bisect`? 

The only culprit I can think of is https://github.com/marmelab/admin-on-rest/pull/1616/files#diff-0c7e48ec3f03db5532bb24babd4872b8, but I don't understand how it would break the feature.

**Comment by afilp** [comment]
008b97e70179a1369f596da759b6b18575c136ab is the first bad commit, based on the bisect.

**Comment by tkvw** [comment]
This could be related to: https://github.com/marmelab/admin-on-rest/pull/1621

**Comment by afilp** [comment]
@tkvw I added your code to the latest `next`, did `make build` and this used it. Unfortunately it had no difference. Do you BTW experience the same problem, if you use a `ReferenceInput Filter` that references the resource used by the `List` itself?

**Comment by tkvw** [comment]
@afilp : Check your redux action log, do you see all the `Fetch` actions?

**Comment by afilp** [comment]
I only get 2 MATCHING_SUCCESS (not 3), if this is what you are asking:

![image](https://user-images.githubusercontent.com/7850073/37236573-bc45b9ca-2411-11e8-9aff-ea4b8932fec9.png)

**Comment by afilp** [comment]
Not sure if this gives any hint, I removed the `{...props}` from the `filter` just to see...

![image](https://user-images.githubusercontent.com/7850073/37236877-288c0f3c-2415-11e8-9cf2-c98425d5667a.png)

...and now the problematic field does get data:

![image](https://user-images.githubusercontent.com/7850073/37236905-6d3d465a-2415-11e8-9b83-a0b1d94e086e.png)

(of course this is just a hack to see how props affected the problem since the display has problems with duplicate select boxes, etc.)

**Comment by afilp** [comment]
Found the cause of the problem, this happens when the filter reference (same as resource) starts as `alwaysOn`:

![image](https://user-images.githubusercontent.com/7850073/37256258-b2cba364-2560-11e8-841f-d480d473a896.png)

If it is not `alwaysOn` and you choose it manually to use it, then the data is retrieved fine (works).

I hope this helps you narrow down the problem in the code.

**Comment by fzaninotto** [comment]
It should be fixed in `next`, thanks to #1621. Can you confirm?

**Comment by afilp** [comment]
I confirm that it works, great work, thanks!

**Comment by fzaninotto** [comment]
Fixed by #1621
